pkg/controller/scheduling/workloadclass.go

```diff
@@ -0,0 +1,56 @@
+package scheduling
+
+import (
+	"context"
+
+	v1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	adminv1 "github.com/acorn-io/acorn/pkg/apis/internal.admin.acorn.io/v1"
+	"github.com/acorn-io/baaah/pkg/router"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// GetWorkloadClassForWorkload determines what WorkloadClass should be used for the given appInstance, container and
+// workload.
+func GetWorkloadClassForWorkload(req router.Request, app *v1.AppInstance, container v1.Container, workload string) (*adminv1.WorkloadClassSpec, error) {
+	workloadClass := app.Status.Defaults.WorkloadClass // Okay to not check as will always be set to new(string) if nil
```

Comment:
This comment is not needed, you are still checking for `nil`-ness below.
```suggestion
	workloadClass := app.Status.Defaults.WorkloadClass
```